doctor/views.py

```diff
@@ -264,6 +261,16 @@ def extract_extension(request) -> HttpResponse:
         mime = magic.from_buffer(content.strip(b"%%EOF\r"), mime=True)
         extension = mimetypes.guess_extension(mime)
 
+    # The extension is still .bin, look in the content if we can infer the
+    # content type
+    if extension == ".bin":
```

Comment:
Couldn't this be part of the `if` statement on 260? Looks like similar kind of mess, according to the comment?

Reply:
the if statement in line 260 only removes whitespaces, for this case we need to search for an specific content

I added it as an extra step because at line 260 we have the chance to get the correct extension, if not we will have to check again if the extension is .bin, instead of having that nested condition i think it looks better if we put it outside on an independent if

Reply:
Sure, but wouldn't your code catch whatever random stuff 261 is trying to catch too? I think we just remove that and replace it with yours.

Reply:
Yes, you are right, I didn't see it that way, both checks are for pdf, we could just remove the old check as you said and keep just the new code and  it should work exactly the same.